lucene/core/src/java/org/apache/lucene/util/fst/FST.java

```diff
@@ -632,36 +641,85 @@ long addNode(Builder<T> builder, Builder.UnCompiledNode<T> nodeIn) throws IOExce
     if (doFixedArray) {
       final int MAX_HEADER_SIZE = 11; // header(byte) + numArcs(vint) + numBytes(vint)
       assert maxBytesPerArc > 0;
-      // 2nd pass just "expands" all arcs to take up a fixed
-      // byte size
+      // 2nd pass just "expands" all arcs to take up a fixed byte size
+
+      // If more than 1/2 of the "slots" would be occupied, write an arc array that may have holes in it
```

Comment:
Is this "1/2" consistent with the condition below (4 * nodeIn.numArcs)? Also, I think it'd be better to have the proportion as at least a constant or (better?) a configurable setting on the builder (not necessarily a constructor attribute).

Reply:
Right it should have said 1/4, and I agree a symbol of some kind is called for. I think I'll add a constant, because rIght now the Builder has no (other) settable members; in practice every setting is passed on the constructor: it's not actually a Builder in the builder-pattern sense, and adding a constructor arg will be kind of intrusive given there is not yet any call for it elsewhere.